ga4gh/cli.py

```diff
@@ -1510,448 +1521,595 @@ def configtest_main(parser=None):
 ##############################################################################
 
 
-def getRawInput(displayString):
-    userResponse = raw_input(displayString)
-    return userResponse
+def getNameFromPath(filePath):
+    """
+    Returns the filename of the specified path without its extensions.
+    This is usually how we derive the default name for a given object.
+    """
+    if len(filePath) == 0:
+        raise ValueError("Cannot have empty path for name")
+    fileName = os.path.split(os.path.normpath(filePath))[1]
+    # We need to handle things like .fa.gz, so we can't use
+    # os.path.splitext
+    ret = fileName.split(".")[0]
+    assert ret != ""
```

Comment:
Are there any characters that will cause us problems downstream? It looks like SQLite stores unicode strings alright, but what about our ID scheme?